### DISCUSSION TITLE
합주실 예약을 위한 카카오 지도 API(Kakao Map API) 사용법

### DISCUSSION THREAD

**Discussion by jaehawilly** [PocketBand/frontend-web#9] [지식자료공유]
# Kakao 지도 API
**Kakao 지도 API** 는 키 발급을 받아야 사용할 수 있습니다.
그리고 키를 발급받기 위해서는 카카오 계정이 필요합니다.

키 발급에는 아래 과정이 필요합니다.

1. [카카오 개발자사이트](https://developers.kakao.com/) (https://developers.kakao.com) 접속
2. 개발자 등록 및 앱 생성
3. 웹 플랫폼 추가: 앱 선택 – [플랫폼] – [Web 플랫폼 등록] – 사이트 도메인 등록
4. 사이트 도메인 등록: [웹] 플랫폼을 선택하고, [사이트 도메인] 을 등록합니다. (예: http://localhost:8080)
5. 페이지 상단의 [JavaScript 키]를 지도 API의 appkey로 사용합니다.
6. 앱을 실행합니다.

**[앱 등록]**

![image](https://github.com/PocketBand/frontend-web/assets/37172636/3004b8ad-cdc4-40fc-b792-acdcb147815a)

**[플랫폼 설정]**

![image](https://github.com/PocketBand/frontend-web/assets/37172636/e55daa87-8274-40be-a8a6-7fe878d7f97f)

ex> http://localhost:8080 로 등록

**Comment by Chominsu98** [comment]
## 카카오 맵에서 상세정보 불러오기 가능 방안 탐색

- 실제 서비스 분석해본 결과 
- 카카오맵은 상세정보 불러올 때 REST방식 GET으로 불러옴 (따라서 직접 호출해볼 경우 JSON 반환)
- 별도 막혀있지 않음(공식 사이트상 제공 api는 아니지만 사용 가능 할 것으로 보임)
- 단지 Proxy로 Cors에러만 피해주면 가능할듯


![Image](https://github.com/PocketBand/frontend-web/assets/37100067/80626402-e106-4786-8ee4-7f098a0361fc)

> 응답값


![Image](https://github.com/PocketBand/frontend-web/assets/37100067/27577e64-542a-40da-89a9-4be382a50984)


### 네이버 맵은 불가

- 네이버맵은 애초에 GraphQl로 엔드포인트가 하나로 파악 불가...

![Image](https://github.com/PocketBand/frontend-web/assets/37100067/56a10208-ef6f-434b-888a-4f005b3b8a1b)

>응답값

![Image](https://github.com/PocketBand/frontend-web/assets/37100067/bf20946f-4b59-4ac0-b547-927e9e766a61)
